### ISSUE TITLE
Does not delete "Kinds/Labels" predicates;

### ISSUE THREAD

**Issue by MichelDiz** [dgraph-io/dgraph#2212]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?
```
Dgraph version   : v1.0.3
Commit SHA-1     : e28847e7
Commit timestamp : 2018-02-08 10:43:50 +1100
Branch           : HEAD
```

- Have you tried reproducing the issue with latest release?
Just in the reported

- What is the hardware spec (RAM, OS)?
 16GB - MacOS (Docker)

- Steps to reproduce the issue (command/config used to run Dgraph).
I have created a lot of users
I tried deleting all at the same time using
```
{
  delete {
    <0x4> ** .
    <0x7> ** .
    <0x9> ** .
    <0xd> ** .
    <0x10> ** .
    <0x13> ** .
    <0x17> ** .
    <0x1a> ** .
    <0x1c> ** .
    <0x1f> ** .
    <0x23> ** .
    <0x25> ** .
    <0x28> ** .

  }
}
```

- Expected behaviour and actual result.

Expected: Delete all edges out of a node. Including the "label/kind/type" predicate.

Actual: Does not delete "Kinds" predicates - When I query for "users(func: has(__User))"
Return all as null

**Comment by pawanrawal** [comment]
Hey @MichelDiz,

Could you try with `v1.0.4` and see if you still see the issue? If you do, then can you share some steps to create the data (set mutations)?

**Comment by MichelDiz** [comment]
Hey man, sorry for the delay.

First of all, has anyone been working on this kind of "bug" since v1.0.3? since I did not even see a related commit then I guess no one worked on something like that. So upgrading to a recent version is almost certain to have no effect.

I'll have to see this tomorrow, but it's as simple as I explained. I think Dgraph forgets to delete predicates that are not listed in Schema. 

By Ratel is ok - Does not return nulls. But when I request a query by dgraph-js it brings me a set of 13 nulls. Since I have made the deletion mutation, there is no other theory than this. Ghosts?

Like I said, I had it deleted. It should not return anything at all through the query "users (func: has (__ User))" if it returns is because Dgraph did not deleted such predicates.

Cheers

**Comment by MichelDiz** [comment]
```
Dgraph version   : v1.0.4
Commit SHA-1     : aad59072
Commit timestamp : 2018-03-09 17:37:44 +1100
Branch           : HEAD
```

Hello again.

I upgraded, deleted it again and Dgraph-JS keeps sending me the same nulls.

I query through Ratel for the created "kind" predicate and it does not return. I believe that Ratel "filters" this first. But the dgraph-js did not.

In my view it seems a mixed problem that relates the dgraph-js and the dgraph itself.

For Dgraph should not send anything to JS like "We got what you query." Since it is deleted, should have nothing to serve. In the sense that he should not identify the 13 exact nodes that were with the kinds/labels.

A gets the question, the problem is more Dgraph or Dgraph-js?

To reproduce the problem, just create a simple JS application with a set of kinds. https://docs.dgraph.io/howto/#giving-nodes-a-type - Then delete them and then search for them again. Using "has (__some)".

**Comment by MichelDiz** [comment]
To avoid null results I'm doing this roundtrip.
But I think this should add a workload to Dgraph, right? anyway if you go investigating this and solving this is my temporary solution until there.
`users(func: has(__User)) @filter( eq(isActive, true) AND (not has(__deleted))) `

**Comment by pawanrawal** [comment]
This is a bug with `has` queries where empty PL's are not ignored.

**Comment by pawanrawal** [comment]
It was fixed by 999a4e0c42b9a10c4111de52eb5703ad682a2bbe.
